src/widget/file/filepicker.js

```diff
@@ -241,6 +255,25 @@ class Filepicker extends Widget {
         }
     }
 
+    _resizeFile( file, mediaType ) {
+        return new Promise( ( resolve, reject ) => {
+            if ( mediaType !== 'image/*' ) {
+                reject( file );
+            }
+
+            // file is image, resize it
+            if ( this.props && this.props.maxPixels ) {
```

Comment:
This function will now not resolve or reject if the condition is not truthy, right (e.g. if maxPixels was not defined in the XForm)? It will remain pending, and `finally()` will never be reached.